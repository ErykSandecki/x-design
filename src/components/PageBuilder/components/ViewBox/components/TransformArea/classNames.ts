import { mapValues } from 'lodash';

// types
import { AnchorResize, AnchorRotate } from 'store/pageBuilder/enums';

export const className = 'TransformArea';

export const classNames = {
  [className]: className,
  anchorResize: {
    name: `${className}__anchor-resize`,
    modificators: {
      ...mapValues(
        AnchorResize,
        (anchor) => `${className}__anchor-resize--${anchor}`,
      ),
    },
  },
  anchorRotate: {
    name: `${className}__anchor-rotate`,
    modificators: {
      ...mapValues(
        AnchorRotate,
        (anchor) => `${className}__anchor-rotate--${anchor}`,
      ),
    },
  },
};
