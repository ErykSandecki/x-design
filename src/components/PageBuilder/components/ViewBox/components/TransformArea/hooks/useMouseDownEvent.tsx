import { MouseEvent, RefObject } from 'react';
import { useDispatch } from 'react-redux';

// store
import { updateEventsStatus } from 'store/pageBuilder/actions';

// types
import { AnchorResize } from 'store/pageBuilder/enums';
import { MouseButton, MouseMode, T2DCoordinates } from 'types';

export type TUseMouseDownEvent = {
  onMouseDownAnchorResize: (anchor: AnchorResize, event: MouseEvent) => void;
};

export const useMouseDownEvent = (
  cursorPosition: RefObject<T2DCoordinates>,
  mouseMode: MouseMode,
): TUseMouseDownEvent => {
  const dispatch = useDispatch();

  const handleMouseDownAchnorResize = (
    anchor: AnchorResize,
    event: MouseEvent,
  ): void => {
    if (event.buttons === MouseButton.lmb && mouseMode === MouseMode.default) {
      event.stopPropagation();

      dispatch(
        updateEventsStatus({ isResizing: true, selectedAnchorResize: anchor }),
      );
      cursorPosition.current = {
        x: Math.round(event.clientX),
        y: Math.round(event.clientY),
      };
    }
  };

  return {
    onMouseDownAnchorResize: handleMouseDownAchnorResize,
  };
};
