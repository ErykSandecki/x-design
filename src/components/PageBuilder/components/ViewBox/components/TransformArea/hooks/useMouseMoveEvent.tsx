import { RefObject, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

// store
import { eventSelectorCreator } from 'store/pageBuilder/selectors';

// types
import { AnchorResize } from 'store/pageBuilder/enums';
import { T2DCoordinates, TElement } from 'types';

// utils
import { handleResizeElement } from '../utils/handleResizeElement';

export type TUseMouseMoveEvent = void;

export const useMouseMoveEvent = (
  cursorPosition: RefObject<T2DCoordinates>,
  height: TElement['height'],
  id: TElement['id'],
  width: TElement['width'],
  x: TElement['coordinates']['x'],
  y: TElement['coordinates']['y'],
): TUseMouseMoveEvent => {
  const anchorResize = useSelector(
    eventSelectorCreator('selectedAnchorResize'),
  );
  const dispatch = useDispatch();

  const handleMouseMoveAnchorResize = (event: MouseEvent): void => {
    event.stopPropagation();
    handleResizeElement(
      cursorPosition,
      dispatch,
      event,
      height,
      id,
      width,
      x,
      y,
    );
  };

  useEffect(() => {
    if (anchorResize !== AnchorResize.none) {
      window.addEventListener('mousemove', handleMouseMoveAnchorResize);
    }

    return () => {
      window.removeEventListener('mousemove', handleMouseMoveAnchorResize);
    };
  }, [anchorResize, cursorPosition.current.x, cursorPosition.current.y]);
};
