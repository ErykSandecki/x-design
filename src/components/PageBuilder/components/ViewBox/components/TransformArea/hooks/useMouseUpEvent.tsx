import { useDispatch } from 'react-redux';
import { useEffect } from 'react';

// store
import { updateEventsStatus } from 'store/pageBuilder/actions';

// types
import { AnchorResize } from 'store/pageBuilder/enums';

export type TUseMouseUpEvent = void;

export const useMouseUpEvent = (): TUseMouseUpEvent => {
  const dispatch = useDispatch();

  const handleMouseUpAnchorResize = (): void => {
    dispatch(
      updateEventsStatus({
        isResizing: false,
        selectedAnchorResize: AnchorResize.none,
      }),
    );
  };

  useEffect(() => {
    window.addEventListener('mouseup', handleMouseUpAnchorResize);

    return () => {
      window.removeEventListener('mouseup', handleMouseUpAnchorResize);
    };
  }, []);
};
