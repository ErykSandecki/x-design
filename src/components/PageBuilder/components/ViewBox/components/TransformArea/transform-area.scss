@import '../../../../../../styles/z-indexes';

$--xd-size: 5px;
$--xd-offset-rotation: 10px;

.TransformArea {
  left: 0;
  pointer-events: none;
  position: absolute;
  top: 0;

  &__anchor-resize {
    pointer-events: all;
    position: absolute;

    &--east {
      cursor: ew-resize;
      height: calc(100% - #{$--xd-size});
      right: calc(-#{$--xd-size} / 2);
      top: calc(#{$--xd-size} / 2);
      width: $--xd-size;
    }

    &--none {
      display: none;
    }

    &--north {
      cursor: ns-resize;
      height: $--xd-size;
      left: calc(#{$--xd-size} / 2);
      top: calc(-#{$--xd-size} / 2);
      width: calc(100% - #{$--xd-size});
    }

    &--north-east {
      cursor: nesw-resize;
      height: $--xd-size;
      right: calc(-#{$--xd-size} / 2);
      top: calc(-#{$--xd-size} / 2);
      width: $--xd-size;
    }

    &--north-west {
      cursor: nwse-resize;
      height: $--xd-size;
      left: calc(-#{$--xd-size} / 2);
      top: calc(-#{$--xd-size} / 2);
      width: $--xd-size;
      z-index: 10;
    }

    &--south {
      bottom: calc(-#{$--xd-size} / 2);
      cursor: ns-resize;
      height: $--xd-size;
      left: calc(#{$--xd-size} / 2);
      width: calc(100% - #{$--xd-size});
    }

    &--south-east {
      bottom: calc(-#{$--xd-size} / 2);
      cursor: nwse-resize;
      height: $--xd-size;
      right: calc(-#{$--xd-size} / 2);
      width: $--xd-size;
    }

    &--south-west {
      bottom: calc(-#{$--xd-size} / 2);
      cursor: nesw-resize;
      height: $--xd-size;
      left: calc(-#{$--xd-size} / 2);
      width: $--xd-size;
    }

    &--west {
      cursor: ew-resize;
      height: calc(100% - #{$--xd-size});
      left: calc(-#{$--xd-size} / 2);
      top: calc(#{$--xd-size} / 2);
      width: $--xd-size;
    }
  }

  &__anchor-rotate {
    cursor: grab;
    height: $--xd-size;
    pointer-events: all;
    position: absolute;
    width: $--xd-size;

    &--north-east {
      right: calc(-#{$--xd-offset-rotation});
      top: calc(-#{$--xd-offset-rotation});
    }

    &--north-west {
      left: calc(-#{$--xd-offset-rotation});
      top: calc(-#{$--xd-offset-rotation});
    }

    &--south-east {
      bottom: calc(-#{$--xd-offset-rotation});
      right: calc(-#{$--xd-offset-rotation});
    }

    &--south-west {
      bottom: calc(-#{$--xd-offset-rotation});
      left: calc(-#{$--xd-offset-rotation});
    }
  }
}
