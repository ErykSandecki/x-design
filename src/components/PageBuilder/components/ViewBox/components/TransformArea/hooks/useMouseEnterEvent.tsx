// types
import { AnchorResize, AnchorRotate } from 'store/pageBuilder/enums';
import { TUseChangeCursor } from 'hooks';

// utils
import { getAnchorResizeCursorAngle } from '../utils/getAnchorResizeCursorAngle';
import { getAnchorRotateCursorAngle } from '../utils/getAnchorRotateCursorAngle';

export type TUseMouseEnterEvent = {
  onMouseEnterAnchorResize: (anchor: AnchorResize) => void;
  onMouseEnterAnchorRotate: (anchor: AnchorRotate) => void;
};

export const useMouseEnterEvent = (
  cursorAngle: number,
  isPressingResize: TUseChangeCursor['isPressing'],
  isPressingRotate: TUseChangeCursor['isPressing'],
  onMouseEnterResize: TUseChangeCursor['onMouseEnter'],
  onMouseEnterRotate: TUseChangeCursor['onMouseEnter'],
): TUseMouseEnterEvent => {
  const handleMouseEnterAchnorResize = (anchor: AnchorResize): void => {
    if (!isPressingRotate) {
      onMouseEnterResize(getAnchorResizeCursorAngle(anchor, cursorAngle), 0);
    }
  };

  const handleMouseEnterAchnorRotate = (anchor: AnchorRotate): void => {
    if (!isPressingResize) {
      const angle = getAnchorRotateCursorAngle(anchor, cursorAngle);
      onMouseEnterRotate(angle, angle - cursorAngle);
    }
  };

  return {
    onMouseEnterAnchorResize: handleMouseEnterAchnorResize,
    onMouseEnterAnchorRotate: handleMouseEnterAchnorRotate,
  };
};
