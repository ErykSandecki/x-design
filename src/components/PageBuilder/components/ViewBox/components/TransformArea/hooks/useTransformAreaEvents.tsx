import { useRef } from 'react';

// hooks
import { TUseMouseDownEvent, useMouseDownEvent } from './useMouseDownEvent';
import { useMouseMoveEvent } from './useMouseMoveEvent';
import { useMouseUpEvent } from './useMouseUpEvent';

// others
import { BASE_2D } from 'shared';

// types
import { MouseMode } from 'types/enums/mouseMode';
import { TElement } from 'types';

export type TUseTransformAreaEvents = TUseMouseDownEvent;

export const useTransformAreaEvents = (
  height: TElement['height'],
  id: TElement['id'],
  mouseMode: MouseMode,
  width: TElement['width'],
  x: TElement['coordinates']['x'],
  y: TElement['coordinates']['y'],
): TUseTransformAreaEvents => {
  const cursorPosition = useRef(BASE_2D);
  const events = useMouseDownEvent(cursorPosition, mouseMode);

  useMouseMoveEvent(cursorPosition, height, id, width, x, y);
  useMouseUpEvent();

  return events;
};
