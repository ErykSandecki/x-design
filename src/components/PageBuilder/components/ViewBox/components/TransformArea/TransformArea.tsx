import { FC } from 'react';
import { kebabCase } from 'lodash';

// components
import { E2EDataAttribute } from 'shared';

// hooks
import { useTransformAreaEvents } from './hooks/useTransformAreaEvents';
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// types
import { AnchorResize, AnchorRotate } from 'store/pageBuilder/enums';
import { E2EAttribute, TElement } from 'types';
import { MouseMode } from 'types/enums/mouseMode';

// styles
import styles from './transform-area.scss';

// utils
import { enumToArray } from 'utils';

export type TTransformAreaProps = {
  height: TElement['height'];
  id: TElement['id'];
  moseMode: MouseMode;
  width: TElement['width'];
  x: TElement['coordinates']['x'];
  y: TElement['coordinates']['y'];
};

const TransformArea: FC<TTransformAreaProps> = ({
  height,
  id,
  moseMode,
  width,
  x,
  y,
}) => {
  const { onMouseDownAnchorResize } = useTransformAreaEvents(
    height,
    id,
    moseMode,
    width,
    x,
    y,
  );
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <div
      className={cx(classNamesWithTheme[className])}
      style={{ height, width }}
    >
      {/* RESIZE AREA */}
      {enumToArray(AnchorResize).map((anchor) => (
        <E2EDataAttribute
          key={anchor as keyof typeof AnchorResize}
          type={E2EAttribute.anchorResize}
          value={kebabCase(anchor as string)}
        >
          <div
            className={cx(
              classNamesWithTheme.anchorResize.name,
              classNamesWithTheme.anchorResize.modificators[
                anchor as keyof typeof AnchorResize
              ],
            )}
            onMouseDown={(event) =>
              onMouseDownAnchorResize(anchor as AnchorResize, event)
            }
          />
        </E2EDataAttribute>
      ))}

      {/* ROTATE AREA */}
      {enumToArray(AnchorRotate).map((anchor) => (
        <E2EDataAttribute
          key={anchor as keyof typeof AnchorRotate}
          type={E2EAttribute.anchorRotate}
          value={kebabCase(anchor as string)}
        >
          <div
            className={cx(
              classNamesWithTheme.anchorRotate.name,
              classNamesWithTheme.anchorRotate.modificators[
                anchor as keyof typeof AnchorRotate
              ],
            )}
            // onMouseDown={(event) =>
            //   events.onMouseDown(anchor as AnchorResize, event)
            // }
          />
        </E2EDataAttribute>
      ))}
    </div>
  );
};

export default TransformArea;
