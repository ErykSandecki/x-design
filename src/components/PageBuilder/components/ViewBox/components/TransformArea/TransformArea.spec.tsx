import { fireEvent, render } from '@testing-library/react';
import { kebabCase } from 'lodash';
import { Provider } from 'react-redux';
import { RefObject } from 'react';

// components
import TransformArea from './TransformArea';

// core
import { RefsProvider } from 'pages/PageBuilderPage/core/RefsProvider';

// hooks
import { useTransformAreaEvents } from './hooks/useTransformAreaEvents';

// mocks
import { selectedElementMock } from 'test/mocks/reducer/pageBuilderMock';
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

// types
import { AnchorResize, AnchorRotate } from 'store/pageBuilder/enums';
import { E2EAttribute } from 'types';
import { MouseMode } from 'types/enums/mouseMode';

// utils
import { createHtmlElement } from 'utils';
import { getByE2EAttribute } from 'test';

const zoomBox = document.createElement('div');

const elementRef = {
  current: createHtmlElement('div'),
} as RefObject<HTMLDivElement>;

const mockCallBack = jest.fn();

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: (): any => mockCallBack,
}));

jest.mock('./hooks/useTransformAreaEvents', () => ({
  useTransformAreaEvents: jest.fn(),
}));

describe('TransformArea snapshots', () => {
  beforeEach(() => {
    (useTransformAreaEvents as jest.Mock).mockReturnValue({
      onMouseDownAnchorResize: mockCallBack,
      onMouseDownAnchorRotate: mockCallBack,
      onMouseEnterAnchorResize: mockCallBack,
      onMouseEnterAnchorRotate: mockCallBack,
      onMouseLeaveAnchorResize: mockCallBack,
      onMouseLeaveAnchorRotate: mockCallBack,
    });
  });

  it('should render TransformArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('TransformArea behaviors', () => {
  beforeEach(() => {
    (useTransformAreaEvents as jest.Mock).mockImplementation(() => ({
      onMouseDownAnchorResize: mockCallBack,
      onMouseDownAnchorRotate: mockCallBack,
      onMouseEnterAnchorResize: mockCallBack,
      onMouseEnterAnchorRotate: mockCallBack,
      onMouseLeaveAnchorResize: mockCallBack,
      onMouseLeaveAnchorRotate: mockCallBack,
    }));
  });

  it('should triger mouse down event anchor resize', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // action
    fireEvent.mouseDown(
      getByE2EAttribute(
        container,
        E2EAttribute.anchorResize,
        kebabCase(AnchorResize.east),
      ),
    );

    // result
    expect(mockCallBack.mock.calls.length).toBe(1);
  });

  it('should triger mouse enter event anchor resize', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // action
    fireEvent.mouseEnter(
      getByE2EAttribute(
        container,
        E2EAttribute.anchorResize,
        kebabCase(AnchorResize.northEast),
      ),
    );

    // result
    expect(mockCallBack.mock.calls.length).toBe(1);
  });

  it('should triger mouse leave event anchor resize', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // action
    fireEvent.mouseLeave(
      getByE2EAttribute(
        container,
        E2EAttribute.anchorRotate,
        kebabCase(AnchorRotate.northEast),
      ),
    );

    // result
    expect(mockCallBack.mock.calls.length).toBe(1);
  });

  it('should triger mouse down event anchor rotate', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // action
    fireEvent.mouseDown(
      getByE2EAttribute(
        container,
        E2EAttribute.anchorRotate,
        kebabCase(AnchorRotate.northEast),
      ),
    );

    // result
    expect(mockCallBack.mock.calls.length).toBe(1);
  });

  it('should triger mouse neter event anchor rotate', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // action
    fireEvent.mouseEnter(
      getByE2EAttribute(
        container,
        E2EAttribute.anchorRotate,
        kebabCase(AnchorRotate.northEast),
      ),
    );

    // result
    expect(mockCallBack.mock.calls.length).toBe(1);
  });

  it('should triger mouse neter event anchor rotate', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <RefsProvider zoomBoxRefHtml={zoomBox}>
          <TransformArea
            counterAngle={0}
            elementRef={elementRef}
            height={100}
            id={selectedElementMock.id}
            moseMode={MouseMode.default}
            angle={0}
            width={100}
            x={100}
            y={100}
          />
        </RefsProvider>
      </Provider>,
    );

    // action
    fireEvent.mouseLeave(
      getByE2EAttribute(
        container,
        E2EAttribute.anchorRotate,
        kebabCase(AnchorRotate.northEast),
      ),
    );

    // result
    expect(mockCallBack.mock.calls.length).toBe(1);
  });
});
