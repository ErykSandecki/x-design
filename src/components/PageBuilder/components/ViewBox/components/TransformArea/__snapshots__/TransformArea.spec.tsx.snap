// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformArea snapshots should render TransformArea 1`] = `
<DocumentFragment>
  <div
    class="TransformArea TransformArea--light"
    style="height: 100px; width: 100px;"
  >
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--east TransformArea__anchor-resize--east--light"
      data-test-anchor-resize="east"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--none TransformArea__anchor-resize--none--light"
      data-test-anchor-resize="none"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--north TransformArea__anchor-resize--north--light"
      data-test-anchor-resize="north"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--northEast TransformArea__anchor-resize--northEast--light"
      data-test-anchor-resize="north-east"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--northWest TransformArea__anchor-resize--northWest--light"
      data-test-anchor-resize="north-west"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--south TransformArea__anchor-resize--south--light"
      data-test-anchor-resize="south"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--southEast TransformArea__anchor-resize--southEast--light"
      data-test-anchor-resize="south-east"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--southWest TransformArea__anchor-resize--southWest--light"
      data-test-anchor-resize="south-west"
    />
    <div
      class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--west TransformArea__anchor-resize--west--light"
      data-test-anchor-resize="west"
    />
    <div
      class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--northEast TransformArea__anchor-rotate--northEast--light"
      data-test-anchor-rotate="north-east"
    />
    <div
      class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--northWest TransformArea__anchor-rotate--northWest--light"
      data-test-anchor-rotate="north-west"
    />
    <div
      class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--southEast TransformArea__anchor-rotate--southEast--light"
      data-test-anchor-rotate="south-east"
    />
    <div
      class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--southWest TransformArea__anchor-rotate--southWest--light"
      data-test-anchor-rotate="south-west"
    />
  </div>
</DocumentFragment>
`;
